test/app-luatest/gh_9806_http_client_version_test.lua

```diff
@@ -0,0 +1,95 @@
+end)
+
+local http_request_script = string.dump(function()
+    local http_client = require('http.client').new()
+    local proto = os.getenv('PROTO')
+    local port = os.getenv('HTTP_SERVER_PORT')
+    local uri = proto .. '://127.0.0.1:' .. port
+    pcall(http_client.get, http_client, uri, {
+        verbose = true,
+        http_version = os.getenv('HTTP_VERSION'),
+    })
+end)
+
+g.test_http_version = function()
+    local dir = treegen.prepare_directory(g, {}, {})
+    local script_name = 'http_request.lua'
+    treegen.write_script(dir, script_name, http_request_script)
+
+    local args = {script_name}
+    local opts = {nojson = true, stderr = true}
+    local env = {HTTP_SERVER_PORT = g.server:name().port}
+    env.PROTO = 'http'
+    env.HTTP_VERSION = nil
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.1')
+    env.HTTP_VERSION = '1.0'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.0')
+    env.HTTP_VERSION = '1.1'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.1')
+    env.HTTP_VERSION = '2'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'Connection: Upgrade, HTTP2-Settings')
+    env.HTTP_VERSION = '2-tls'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.1')
+    env.HTTP_VERSION = '2-prior-knowledge'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, '[HTTP/2] [1] OPENED stream')
+end
+
+g.test_https_version = function()
+    local dir = treegen.prepare_directory(g, {}, {})
+    local script_name = 'http_request.lua'
+    treegen.write_script(dir, script_name, http_request_script)
+
+    local args = {script_name}
+    local opts = {nojson = true, stderr = true}
+    local env = {HTTP_SERVER_PORT = g.server:name().port}
+    env.PROTO = 'https'
+    env.HTTP_VERSION = nil
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+    env.HTTP_VERSION = '1.0'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers http/1.1')
+    env.HTTP_VERSION = '1.1'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers http/1.1')
+    env.HTTP_VERSION = '2'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+    env.HTTP_VERSION = '2-tls'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+    env.HTTP_VERSION = '2-prior-knowledge'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+end
```

Comment:
Please rewrite testcases using luatest parameterization, see example here - https://github.com/tarantool/tarantool/blob/58f5ffe43bffa0901d8d577bbb4d523c4e7e682a/test/app-luatest/uri_unit_test.lua#L11C7-L33

About parameterization in luatest doc: https://github.com/tarantool/luatest/?tab=readme-ov-file#parametrization

the main idea - to split test data and test code

Reply:
Rewrited testcases using luatest parametrization. Met some troubles with `group()` function from `luatest`. If `res` string contains symbol '/' test-run.py can't run this test and gets an error in `group()` function. Asked Oleg about this issue and waiting for an answer tomorrow. 
Could you check the rest of the testcases? In comments I added strings that I want to use for checking. If current strings are enough I can leave them and delete comments.

Reply:
Seems mostly fine, but left minor comments